tests/unit/setup-application-context-test.js

```diff
@@ -51,10 +55,20 @@ module('setupApplicationContext', function(hooks) {
           throw new Error('Model hook error from /widgets');
         },
       }),
+      'route:slow': Route.extend({
+        model() {
+          assert.step('model start');
+          return new Promise(resolve => {
+            setTimeout(() => {
+              assert.step('model resolving');
+              resolve();
+            }, 50);
+          });
+        },
+      }),
     });
 
-    setApplication(application);
-
+    setContext(this);
```

Comment:
@rwjblue This didn't seem to fix the test issues @dexturr and I were seeing (when we added the shared state workaround discussed [here](https://github.com/emberjs/ember-test-helpers/pull/482#discussion_r236343138)). We weren't quite sure why it didn't resolve the issue?

Reply:
@rwjblue ping

We were wondering if order dependent tests were failing [due to sharing an Ember app across all tests](https://github.com/emberjs/ember-test-helpers/blob/master/tests/helpers/resolver.js#L11)? But were unsure of how to proceed to discover to fix this bug.

Would love to get in there and get it sorted! Any ideas?

Cheers.

Reply:
Ya, you were totally correct. We had some test leakage issues, but not in the most obvious way 😩.

I documented my findings in d115c82bce552de9d2647fab6125efa786e85e0d along with  some better/alternative fixes, though I think the fix in that commit is fine for now.